android/src/play/java/com/dooboolab/RNIap/RNIapModule.kt

```diff
@@ -69,16 +69,26 @@ class RNIapModule(
                             promise.safeReject(PromiseUtils.E_NOT_PREPARED, "Unable to auto-initialize connection")
                         }
                     } else {
+                        promise.safeReject(PromiseUtils.E_UNKNOWN, "ensureConnection - incorrect parameter in resolve")
                         Log.i(TAG, "Incorrect parameter in resolve")
                     }
                 },
                 {
-                    if (it.size > 1 && it[0] is String && it[1] is String) {
+                    var errorCode: String? = null
+                    var errorMessage: String? = null
+                    if (it.isNotEmpty() && it[0] is WritableNativeMap) {
+                        val errorMap = it[0] as WritableNativeMap
+                        errorCode = errorMap.getString("code")
+                        errorMessage = errorMap.getString("message")
+                    }
+
+                    if (errorCode is String && errorMessage is String) {
                         promise.safeReject(
-                            it[0] as String,
-                            it[1] as String
+                            errorCode,
+                            errorMessage
                         )
                     } else {
```

Comment:
In this block, Promise is not neither rejected nor resolved. So, when this block is called, any RNIap function is awaiting infitinitely. We have to reject the promise in this case